src/myst/syntax/parser.cr

```diff
@@ -1039,10 +1039,20 @@ module Myst
     end
 
     def parse_magic_constant
-      start = expect(Token::Type::MAGIC_CONST)
-      skip_space
-
-      return MagicConst.from(start.value).at(start.location)
+      case (token = current_token).type
+      when Token::Type::MAGIC_FILE
+        expect(Token::Type::MAGIC_FILE)
+        skip_space
+        MagicConst.new(:"__FILE__").at(token.location)
+      when Token::Type::MAGIC_LINE
+        expect(Token::Type::MAGIC_LINE)
+        skip_space
+        MagicConst.new(:"__LINE__").at(token.location)
+      else
+        expect(Token::Type::MAGIC_DIR)
+        skip_space
+        MagicConst.new(:"__DIR__").at(token.location)
+      end
```

Comment:
i don’t know if this would be better, but you should be able to `expect` with multiple token types, then `case` on the token that comes from that.

then you wouldn’t have to have `skip_space` repeated in each clause.